cypress/component/utils/bucket_utils.spec.js

```diff
@@ -398,92 +397,99 @@ const match_results = [
-    const missingDataUse = find(b => !b.isRP && isUndefined(b.dataUse))(buckets);
-    expect(missingDataUse).to.not.be.empty;
-    expect(missingDataUse.datasets).to.not.be.empty;
-    expect(missingDataUse.datasets.length).to.eq(1);
-    expect(missingDataUse.dataUse).to.be.undefined;
-    expect(missingDataUse.dataUses).to.be.empty;
+    cy.wrap(binCollectionToBuckets(dar_collection)).then((buckets) => {
+      const missingDataUse = buckets.find(b => !b.isRP && isUndefined(b.dataUse));
+      cy.wrap(missingDataUse).should('not.be.empty');
+      cy.wrap(missingDataUse.datasets).should('not.be.empty');
+      cy.wrap(missingDataUse.datasets.length).should('eq', 1);
+      cy.wrap(missingDataUse.dataUse).should('be.undefined');
+      cy.wrap(missingDataUse.dataUses).should('be.empty');
+    });
   });
 
-  it('buckets should be filtered to datasets containing one dac id: 1', async () => {
+  it('buckets should be filtered to datasets containing one dac id: 1', () => {
     cy.stub(Match, 'findMatchBatch').returns(match_results);
-    const buckets = await binCollectionToBuckets(dar_collection, [1]);
-    const dataAccessBuckets = filter(b => !b.isRP)(buckets);
-    expect(dataAccessBuckets).to.exist;
-    expect(dataAccessBuckets.length).to.eq(1);
-    expect(dataAccessBuckets[0].datasetIds.length).to.eq(1);
-    forEach(b => {
-      forEach(e => {
-        expect(b.datasetIds).to.contain(e.datasetId);
-      })(b.elections);
-    })(buckets);
+    cy.wrap(binCollectionToBuckets(dar_collection, [1])).then((buckets) => {
+      const dataAccessBuckets = buckets.filter(b => !b.isRP);
+      cy.wrap(dataAccessBuckets).should('exist');
+      cy.wrap(dataAccessBuckets.length).should('eq', 1);
+      cy.wrap(dataAccessBuckets[0].datasetIds.length).should('eq', 1);
+      forEach(b => {
+        forEach(e => {
+          cy.wrap(b.datasetIds).should('contain', e.datasetId);
+        })(b.elections);
+      })(buckets);
+    });
   });
 
-  it('buckets should be filtered to datasets containing two dac ids: 1 & 5', async () => {
+  it('buckets should be filtered to datasets containing two dac ids: 1 & 5', () => {
     cy.stub(Match, 'findMatchBatch').returns(match_results);
-    const buckets = await binCollectionToBuckets(dar_collection, [1, 5]);
-    const dataAccessBuckets = filter(b => !b.isRP)(buckets);
-    expect(dataAccessBuckets).to.exist;
-    expect(dataAccessBuckets.length).to.eq(2);
-    forEach(b => {
-      forEach(e => {
-        expect(b.datasetIds).to.contain(e.datasetId);
-      })(b.elections);
-    })(buckets);
+    cy.wrap(binCollectionToBuckets(dar_collection, [1, 5])).then((buckets) => {
+      const dataAccessBuckets = buckets.filter(b => !b.isRP);
+      cy.wrap(dataAccessBuckets).should('exist');
+      cy.wrap(dataAccessBuckets.length).should('eq', 2);
+      forEach(b => {
+        forEach(e => {
+          cy.wrap(b.datasetIds).should('contain', e.datasetId);
+        })(b.elections);
+      })(buckets);
```

Comment:
I think these tests have issues too, if I do `buckets.forEach` instead. the tests fail.

Reply:
I created https://broadworkbench.atlassian.net/browse/DT-1816 for this issue.